planner/core/integration_test.go

```diff
@@ -7823,3 +7823,14 @@ func TestNullConditionForPrefixIndex(t *testing.T) {
 		"  └─StreamAgg_9 1.00 cop[tikv]  funcs:count(1)->Column#7",
 		"    └─IndexRangeScan_17 99.90 cop[tikv] table:t1, index:idx2(c1, c2) range:[\"0xfff\" -inf,\"0xfff\" +inf], keep order:false, stats:pseudo"))
 }
+
+func TestAutoIncrementCheckWithCheckConstraint(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec(`CREATE TABLE t (
+		id INTEGER NOT NULL AUTO_INCREMENT,
+		CHECK (id IN (0, 1)),
+		KEY idx_autoinc_id (id)
```

Comment:
Is this KEY needed?

Reply:
Yes. The `AUTO_INCREMENT` column has to be defines as a key (any kind is fine).